eblob_kit.py

```diff
@@ -1064,11 +1079,15 @@ def recover_index(data, destination):
                           offset)
             break
 
-    def recover_blob(self, destination):
+    def recover_blob(self, destination, overwrite=False):
         """Recover blob from data."""
         basename = os.path.basename(self._blob.data.path)
         blob_path = os.path.join(destination, basename)
-        blob = Blob.create(blob_path)
+
+        if not is_destination_writable(self._blob.data.path,  blob_path, overwrite):
+            raise RuntimeError("can't recover to already existing blob file: {}".format(blob_path))
+
+        blob = Blob.create(path=blob_path, is_sorted_index=self._blob.index.sorted)
```

Comment:
Why was `is_sorted_index=self._blob.index.sorted` added?

Reply:
If we know in advance that it would be sorted index written on recovery, it is good checkpoint here to mark index files suffixes while creating blob. It was _jonny-k_ suggestion. 

Should it also be in separate commit?

Reply:
> If we know in advance that it would be sorted index written on recovery, it is good checkpoint here to mark index files suffixes while creating blob. It was _jonny-k_ suggestion.

`self._blob.index.sorted` generally doesn't mean that recovered index will be sorted.

> Should it also be in separate commit?

Yes, both this and adding `is_sorted_index` to `Blob.create()` should be done in a separate commit.